src/data/nat/factorization.lean

```diff
@@ -86,27 +88,79 @@ by simp [factorization, add_equiv.map_eq_zero_iff, multiset.coe_eq_zero]
 /-- For nonzero `a` and `b`, the power of `p` in `a * b` is the sum of the powers in `a` and `b` -/
 @[simp] lemma factorization_mul {a b : ℕ} (ha : a ≠ 0) (hb : b ≠ 0) :
   (a * b).factorization = a.factorization + b.factorization :=
-by { ext p, simp only [add_apply, factorization_eq_count,
-  count_factors_mul_of_pos (zero_lt_iff.mpr ha) (zero_lt_iff.mpr hb)] }
+by { ext p, simp only [add_apply, ←factors_count_eq,
+                       perm_iff_count.mp (perm_factors_mul ha hb) p, count_append] }
 
 /-- For any `p`, the power of `p` in `n^k` is `k` times the power in `n` -/
-lemma factorization_pow {n k : ℕ} :
+@[simp] lemma factorization_pow (n k : ℕ) :
   factorization (n^k) = k • n.factorization :=
-by { ext p, simp [factorization_eq_count, factors_count_pow] }
+begin
+  induction k with k ih,
+  { simp },
+  rcases n.eq_zero_or_pos with rfl | hn,
+  { simp },
+  rw [pow_succ, factorization_mul hn.ne' (pow_ne_zero _ hn.ne'), ih, succ_eq_one_add, add_smul,
+    one_smul],
+end
+
+/-- For coprime `a` and `b`, the power of `p` in `a * b` is the sum of the powers in `a` and `b` -/
+lemma count_factors_mul_of_coprime {p a b : ℕ} (hab : coprime a b)  :
```

Comment:
Shouldn't these names be about `factorization`?

Reply:
Thanks for the catch! Amended